### ISSUE TITLE
[docs] RandomResizeCrop signature quirks

### ISSUE THREAD

**Issue by vadimkantorov** [pytorch/vision#3763]
1. Why do we need to have so much precision: `1.333333333333`? `1.3` would suffice and be more readable
2. `<InterpolationMode.BILINEAR: 'bilinear'>`. It's not clear from this if the enum should be preferred or the constant. It's also not clear which type is used for InterpolationMode, especially given that for a long time only PIL.Image.InterpolationMode was supported, hence even new codebases are using the legacy PIL types: https://github.com/facebookresearch/dino/blob/58aabc0/main_dino.py#L420
3. Why does `InterpolationMode` live in torchvision.transforms.functional whereas the type using it lives in torchvision.transforms? This enum is basic enough to deserve placement in torchvision.transforms directly IMHO.

![image](https://user-images.githubusercontent.com/1041752/116885862-aa412f00-ac28-11eb-92ad-dffc292c8c51.png)

**Comment by vadimkantorov** [comment]
Also, string constant (despite ths string constant in the signature may be suggestive) does not work. Maybe it should work? That's the interface supported by the underlying `F.interpolate`?

**Comment by NicolasHug** [comment]
@vadimkantorov thanks for the report, please feel free to open a PR to fix these (except for 3 maybe, which should be discussed in a different issue / PR since it's not related to the docs)

**Comment by vadimkantorov** [comment]
I think (1) and (2) actually are about configuration of Sphinx / rendering ...

**Comment by NicolasHug** [comment]
Yes, and they are still doc-related. My point is that 3) is about the API, and will require more discussions than the rest of the items: we can't move stuff around easily when they're public. So it's best not to address all of your points together in the same PR

**Comment by vadimkantorov** [comment]
I propose to also have strings supported as interpolation mode (as Kornia does)...

**Comment by datumbox** [comment]
@vadimkantorov The use of "enum-style" modes was a deliberate choice in order to make the code easier to read/navigate/refactor.

**Comment by fmassa** [comment]
@vadimkantorov about points 3 and 4:
The enum is visible from `transforms.py` in  https://github.com/pytorch/vision/blob/00ea88c0a7ba94bf019730540a51c2577e740f07/torchvision/transforms/transforms.py#L17 and is needed to be in `functional.py` because `transforms.py` import `functional.py`, otherwise we would have cyclic dependencies (except if we extract it to its own file, which I would be fine with).

About supporting `strings` for `InterpolationMode`, I remember discussing this with @datumbox back then and he mentioned that Enums were better for discoverability, specially with smart IDEs that provide auto-completion features.

About points 1 and 2, we should improve the documentation there, and maybe provide a better `__repr__` to `InterpolationMode`

How do you think we could improve the docs so that new projects can know which version they should be using?

**Comment by vadimkantorov** [comment]
Probably also to just be supporting exactly the same interface as F.interpolate does, to not force users have different interoplate mode typing in different parts of pytorch

**Comment by fmassa** [comment]
> These interpolation modes naming are quite standard across opencv / pil / kornia

Well, each of those libraries have different names / conventions for each method, but both PIL and OpenCV use some sort of enums (and not strings) for representing the mode.
From this perspective, using a proper Python enum seems adapted to me. But I would be ok making it support a string as well (which is what [TF does for `tf.image.resize`](https://www.tensorflow.org/api_docs/python/tf/image/resize) )

> So some clearness / examples would be great - and copy-pasteability, that is the enum type should be fully qualified (despite being verbose)

That sounds ok to me.

> Probably also to just be supporting exactly the same interface as F.interpolate does, to not force users have different interoplate mode typing in different parts of pytorch

This is something which is worth further discussion. Now that torchvision transforms support both Tensor and PIL, there is indeed a question on wether we should follow PIL (BC) or lean towards PyTorch.

IMO, the current situation with PyTorch interpolate is a bit messy, with interpolation modes being `linear` / `bilinear` / `trilinear` / etc, while only one single option actually works for a given input shape. I would lean towards not following what PyTorch does wrt that, but instead focus on providing more generic abstractions in PyTorch for interpolating over any arbitrary number of dimensions, which is what I proposed in https://github.com/pytorch/pytorch/issues/10482 and that we are slowly getting there by unifying the CPU implementations of all those methods, see https://github.com/pytorch/pytorch/pull/54500
